stories/Button/Button.tsx

```diff
@@ -1,31 +1,123 @@
+   */
+  label?: string;
+  /**
+   * Indica que el botón esta deshabilitad, esto previene que se pueda hacer click en el botón.
+   */
+  disabled?: boolean;
+  /**
+   * Indica que el botón esta cargando, esto previene que se pueda hacer click en el botón.
+   */
+  loading?: boolean;
+  /**
+   * Tipo de botón, en este caso por defecto es "button".
+   */
+  type: "submit" | "reset" | "button";
+  /**
+   * El icono para mostrar en el botón. Por defecto se muestra a la izquierda del texto (si este existe).
+   */
+  icon?: React.ReactNode;
   /**
-   * Button contents
+   * El icono para mostrar en el botón. Por defecto se muestra a la derecha del texto (si este existe).
    */
-  label: string;
+  rightIcon?: React.ReactNode;
   /**
-   * Optional click handler
+   * Borde redondeado del botón. Por defecto es "md".
    */
-  onClick?: () => void;
+  rounded?: "base" | "md" | "lg";
+  /**
+   * Click handler requerido para el botón
+   */
+  onClickHandler: (event: MouseEvent<HTMLButtonElement>) => void;
+
+  children?: React.ReactNode;
+
+  // TODO: These two are only for the storybook. Remove it when we have a proper styles for it.
+  /**
+   * Color de fondo de botón
+   */
+  backgroundColor?: string;
+  /**
+   * Color de texto de botón
+   */
+  color?: string;
 }
 
-/**
- * Primary UI component for user interaction
- */
-export const Button = ({ backgroundColor, label, ...props }: ButtonProps) => {
+export const Button = ({
+  ariaLabel,
+  backgroundColor,
+  children,
+  color,
+  disabled,
+  icon,
+  label,
+  loading,
+  onClickHandler,
+  rightIcon,
+  rounded = "base",
+  size = "medium",
+  type = "button",
+  variant = "primary",
+  ...props
+}: ButtonProps) => {
+  const mode = {
+    primary: "bg-[#667080] text-[#FFFFFF] active:bg-[#5d6572]",
+    secondary: "bg-[#eef1f4] text-[#667080] active:bg-[#fff1f4]",
+    outlined:
+      "bg-transparent ring-1 ring-[#667080] text-[#667080] active:bg-[#F5F5F5]",
+    icon: "bg-transparent text-[#667080] py-1 px-2 active:bg-[#F5F5F5]",
+  };
+
+  const button = {
+    small: "px-4 py-2 text-xs",
+    medium: "px-5 py-2 text-sm",
+    large: "px-6 py-3 text-base",
+    stretched: "w-full px-6 py-3 text-base",
+  };
+
+  const disabledStyles = disabled
+    ? "disabled:cursor-not-allowed disabled:bg-[#b3b8c0] disabled:text-[#FFFFFF]"
+    : "";
+
+  const focusStyles =
+    "focus:outline-2 focus:outline-offset-2 focus:outline-dashed focus:outline-neutral-400";
+
+  const borderRadius = {
+    base: "rounded",
+    md: "rounded-md",
+    lg: "rounded-lg",
+  };
+
   return (
     <button
-      type="button"
-      className={`rounded p-4 ${backgroundColor}`}
+      aria-disabled={loading || disabled}
+      aria-label={ariaLabel}
+      className={` align-middle font-medium ${borderRadius[rounded]} ${focusStyles} ${mode[variant]} ${button[size]} ${disabledStyles}`}
+      disabled={loading || disabled}
+      style={{ backgroundColor, color }}
```

Comment:
Tratemos de no usar el tag style, teniendo Tailwind no deberia ser necesario.

Reply:
Sip, esto estaba planeado para eliminarse. Enseguida lo hago
https://github.com/frontendcafe/water-call-for-papers/blob/e69a60527361e9eee9cc26c77e0071fc49cb7bda/stories/Button/Button.tsx#L51